src/shared/sam/activation.ts

```diff
@@ -123,6 +123,32 @@ async function registerServerlessCommands(ctx: ExtContext): Promise<void> {
     )
 }
 
+async function activateCodeLensRegistry(context: ExtContext) {
+    try {
+        const registry = new CodelensRootRegistry()
+        ext.codelensRootRegistry = registry
+        await registry.addWatchPattern(pyLensProvider.PYTHON_BASE_PATTERN)
+        await registry.addWatchPattern(jsLensProvider.JAVASCRIPT_BASE_PATTERN)
+        await registry.addWatchPattern(csLensProvider.CSHARP_BASE_PATTERN)
+        await registry.addWatchPattern(goLensProvider.GO_BASE_PATTERN)
+        await registry.addWatchPattern(javaLensProvider.GRADLE_BASE_PATTERN)
+        await registry.addWatchPattern(javaLensProvider.MAVEN_BASE_PATTERN)
+    } catch (e) {
+        vscode.window.showErrorMessage(
+            localize(
+                'AWS.codelens.failToInitializeCode',
+                'Failed to activate Lambda handler {0}',
+                getIdeProperties().codelenses
+            )
+        )
+        getLogger().error('Failed to activate codelens registry', e)
+        // This prevents us from breaking for any reason later if it fails to load. Since
+        // Noop watcher is always empty, we will get back empty arrays with no issues.
+        ext.codelensRootRegistry = new NoopWatcher() as unknown as CodelensRootRegistry
```

Comment:
Leaving the `NoopWatcher` but pretty sure the `NoopWatcher` doesn't need to be here anymore. Might as well guard against the constructor failing I suppose.